apps/app/components/playground/broadcast.tsx

```diff
@@ -76,32 +70,30 @@ const StatusMonitor = ({
     } else if (state.status !== "live") {
       liveEventSentRef.current = false;
     }
-  }, [state.status, streamId, pipelineId, pipelineType, user?.id]);
+  }, [state.status, stream?.id, pipeline?.id, pipeline?.type, user?.id]);
 
   return null;
 };
 
-export function BroadcastWithControls({
-  ingestUrl,
-  className,
-  onCollapse,
-  isCollapsed,
-  streamId,
-  pipelineId,
-  pipelineType,
-}: {
-  ingestUrl: string | null;
-  className?: string;
-  onCollapse?: (collapsed: boolean) => void;
-  isCollapsed?: boolean;
-  streamId?: string;
-  pipelineId?: string;
-  pipelineType?: string;
-}) {
+interface BroadcastUIState {
+  collapsed: boolean;
+  setCollapsed: (value: boolean) => void;
+  toggleCollapsed: () => void;
+}
+
+export const useBroadcastUIStore = create<BroadcastUIState>((set, get) => ({
+  collapsed: false,
+  setCollapsed: value => set({ collapsed: value }),
+  toggleCollapsed: () => set({ collapsed: !get().collapsed }),
```

Comment:
I'm bit new to zustand but I think we can pass prev state in set directly?
https://zustand.docs.pmnd.rs/getting-started/introduction#first-create-a-store
```suggestion
  toggleCollapsed: () => set((state) => ({ collapsed: !state.collapsed })),
```

Reply:
I agree, that should be safer in terms of race conditions.